src/linalg.jl

```diff
@@ -105,7 +105,7 @@ function LinearAlgebra.svd(B::BlockDiagonal; full::Bool=false)::SVD
 end
 
 function LinearAlgebra.cholesky(B::BlockDiagonal)::Cholesky
-    C = BlockDiagonal(map(b -> cholesky(b).U, blocks(B)))
+    C = BlockDiagonal(map(b -> cholesky(b).U.data, blocks(B)))
```

Comment:
i think `parent` is the public API for this

https://github.com/JuliaLang/julia/blob/4015e0d2cdb77e55551ab3e7c31aacf283fc9dcf/stdlib/LinearAlgebra/src/triangular.jl#L160

```suggestion
    C = BlockDiagonal(map(b -> parent(cholesky(b).U), blocks(B)))
```